aikau/src/main/resources/alfresco/dialogs/AlfDialog.js

```diff
@@ -181,6 +181,56 @@ define(["dojo/_base/declare",
       },
 
       /**
+       * Calculates various heights that are used to set dimensions of the dialog when it is created
+       * as well as on resize events (such as resizing the window).
+       * 
+       * @instance
+       * @returns {object} heights The calculated heights
+       * @returns {number} heights.clientHeight The height of the viewport
+       * @returns {number} heights.documentHeight The height of the document
+       * @returns {number} heights.maxBodyHeight The maximum height allowed for the body of the dialog
+       * @returns {number} heights.paddingAdjustment The pixels to allow for padding in the dialog body
+       * @returns {number} heights.simplePanelHeight The height of the [SimplePanel]{@link module:alfresco/layout/SimplePanel} containing the dialog content
+       * @since 1.0.36
+       */
+      calculateHeights: function alfresco_dialogs_AlfDialog__calculateHeights() {
+         var calculatedHeights = {};
+         var docHeight = $(document).height(),
+             clientHeight = $(window).height();
+         var h = (docHeight < clientHeight) ? docHeight : clientHeight;
+         var maxHeight = h - 200;
+
+         var paddingAdjustment = 24;
```

Comment:
Why 24? Arbitrary numbers in code make it very difficult to revisit the code and understand what's happening.